contracts/secret-storage/TeeManager.sol

```diff
@@ -0,0 +1,600 @@
+        uint256 maxAge,
+        IERC20 _stakingToken,
+        uint256 _minStakeAmount,
+        uint256 _slashPercentInBips,
+        uint256 _slashMaxBips
+    ) AttestationAutherUpgradeable(attestationVerifier, maxAge) {
+        _disableInitializers();
+
+        if (address(_stakingToken) == address(0)) revert TeeManagerZeroAddressStakingToken();
+        if (_minStakeAmount == 0) revert TeeManagerZeroMinStakeAmount();
+
+        STAKING_TOKEN = _stakingToken;
+        MIN_STAKE_AMOUNT = _minStakeAmount;
+
+        SLASH_PERCENT_IN_BIPS = _slashPercentInBips;
+        SLASH_MAX_BIPS = _slashMaxBips;
+    }
+
+    //-------------------------------- Overrides start --------------------------------//
+
+    /// @inheritdoc ERC165Upgradeable
+    function supportsInterface(
+        bytes4 interfaceId
+    ) public view virtual override(ERC165Upgradeable, AccessControlUpgradeable) returns (bool) {
+        return super.supportsInterface(interfaceId);
+    }
+
+    /// @inheritdoc UUPSUpgradeable
+    function _authorizeUpgrade(address /*account*/) internal view override onlyRole(DEFAULT_ADMIN_ROLE) {}
+
+    //-------------------------------- Overrides end --------------------------------//
+
+    //-------------------------------- Initializer start --------------------------------//
+
+    /// @notice Thrown when the provided admin address is zero.
+    error TeeManagerZeroAddressAdmin();
+
+    /**
+     * @dev Initializes the contract with the given admin and enclave images.
+     * @param _admin The address of the admin.
+     * @param _images Array of enclave images to initialize.
+     */
+    function initialize(address _admin, EnclaveImage[] memory _images) public initializer {
+        if (_admin == address(0)) revert TeeManagerZeroAddressAdmin();
+
+        __Context_init_unchained();
+        __ERC165_init_unchained();
+        __AccessControl_init_unchained();
+        __UUPSUpgradeable_init_unchained();
+        __AttestationAuther_init_unchained(_images);
+
+        _grantRole(DEFAULT_ADMIN_ROLE, _admin);
+    }
+
+    //-------------------------------- Initializer end --------------------------------//
+
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    IERC20 public immutable STAKING_TOKEN;
+
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    uint256 public immutable MIN_STAKE_AMOUNT;
+
+    /// @notice an integer in the range 0-10^6
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    uint256 public immutable SLASH_PERCENT_IN_BIPS;
+
+    /// @notice expected to be 10^6
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    uint256 public immutable SLASH_MAX_BIPS;
+
+    bytes32 public constant JOBS_ROLE = keccak256("JOBS_ROLE");
+
+    Executors public EXECUTORS;
+
+    SecretStore public SECRET_STORE;
+
+    //-------------------------------- TeeManager start ----------------------------------//
+
+    modifier onlySecretStore() {
```

Comment:
Current approach is better as we already have the SECRET_STORE storage var. With onlyRole, we will have to do an extra transaction to grant the role.

Reply:
what's the pros of using onlyRole that we are missing here? Any security concerns? Also, you can set the ROLE during initialization of contract.

Reply:
TeeManager contract will be deployed first and then the Executors and SecretStore contracts. So, we cannot assign the role during initialization. I have made setter functions for these two contracts.
We can grant the roles within these setter functions but there isn't any security threat with the existing approach.

Reply:
Also, just in case we have to update the Executors or SecretStore contract, first we will have to revoke the role and then grant it to the new contract, so another overhead.